### PR TITLE
v3 53: Add breadcrumbs

### PR DESCRIPTION
**Pull request by mattbowen-usds**
This branch revises the infrastructure setup in mattbowen-usds/v3-51-get-started-round1 to:

- Add breadcrumb data to page requests and a breadcrumb component 
- Fix type annotations throughout
- Run a formatter on pretty much all the code.
- Adjusts the dropdowns on desktop so only one is ever open at a time

### CODE REVIEW

File: tech-far-hub/gatsby-node.ts [hunk 1]
```diff
@@ -65,22 +86,59 @@ export const createPages: GatsbyNode["createPages"] = async ({ graphql, actions,
...
 
     // TODO: Make page-specific templates more general
-    const templateName = pageName === "index" ? "template-index.tsx" : "template-default.tsx";
-    const pagePath = pageName == "index" ? `${parentDirs}/` : `${parentDirs}/${pageIdentier}/`;
+    const templateName = node.pageName === "index" ? "template-index.tsx" : "template-default.tsx";
+    const pagePath = node.pagePath;
     const template = path.resolve("src", "pages", contentType, templateName);
+    const pathParts = node.pagePath.replace(/\/$/, "").split("/");
+    const breadCrumbPaths = pathParts.reduce(
+      (accumulator: string[], currentValue: string) => [
```
Comment (vim-usds) [inline tech-far-hub/gatsby-node.ts:122]: While typing will catch most things, if unit testing is added, I might abstract this reducer's callback function into a separate function to allow that function to have its own test.

Comment (vim-usds) [reply tech-far-hub/gatsby-node.ts:122]: That being said, nice use of chaining and keeping it functional!

File: tech-far-hub/package.json [hunk 2]
```diff
@@ -29,6 +29,7 @@
     "gatsby-transformer-sharp": "^5.0.0",
     "react": "^18.2.0",
     "react-dom": "^18.2.0",
+    "react-use": "^17.4.0",
```
Comment (vim-usds) [inline tech-far-hub/package.json:32]: My favorite utility lib!

File: tech-far-hub/src/pages/get-started/template-default.tsx [hunk 3]
```diff
@@ -2,13 +2,13 @@ import * as React from "react";
 import type { HeadFC, PageProps } from "gatsby";
 import { graphql } from "gatsby";
 import Layout from "../../components/layout";
+import { IPageContext } from "../../types";
 
-const LifecyclePage: React.FC<PageProps<Queries.GetStartedInnerPageQuery>> = ({
-  data,
-  children,
-}: PageProps<Queries.GetStartedInnerPageQuery>) => {
+type LifecyclePageProps = PageProps<Queries.GetStartedInnerPageQuery, IPageContext>;
```
Comment (vim-usds) [inline tech-far-hub/src/pages/get-started/template-default.tsx:7]: One of these you'll have to explain this to me! Nice work!

**Review by vim-usds — APPROVED** [review]
Matt - amazing work!! Very motivating!

File: tech-far-hub/gatsby-node.ts (hunk 1 above)
Comment (mattbowen-usds) [reply tech-far-hub/gatsby-node.ts:122]: Good call --- I'll have to go back and add tests now that we're starting to solidify what we're building.

File: tech-far-hub/package.json (hunk 2 above)
Comment (mattbowen-usds) [reply tech-far-hub/package.json:32]: I learned about it borrowing code from j40 :D

File: tech-far-hub/src/pages/get-started/template-default.tsx (hunk 3 above)
Comment (mattbowen-usds) [reply tech-far-hub/src/pages/get-started/template-default.tsx:7]: Gladly. The short version is PageProps is a generic that can be specified across a couple  parameters. So in this case, I'm passing both the query type and the stuff i expect to get in the PageContext. It also takes a custom type for LocationState and ServerData, but I don't pass those in yet (maybe never tbh).

**Pull request merged**